Assignment1/Question1.java

```diff
@@ -0,0 +1,130 @@
+package Assignment1;
+
+import java.util.HashMap;
+
+public class Question1 {
+
+	public static void main(String[] args) {
+		String s1 = "";
+		String s2 = "";
+		System.out.println(s1 + "  " + s2 + " => " + checkPermutation(s1, s2) + "\n");
+
+		String s3 = "listen";
+		String s4 = "siLenT";
+		System.out.println(s3 + "  " + s4 + " => " + checkPermutation(s3, s4) + "\n");
+
+		String s5 = "Apple";
+		String s6 = "Pabble";
+		System.out.println(s5 + "  " + s6 + " => " + checkPermutation(s5, s6) + "\n");
+
+		String s7 = null;
+		String s8 = null;
+		System.out.println(s7 + "  " + s8 + " => " + checkPermutation(s7, s8));
+	}
+
+	/**
+	 * Checks if the given input is invalid: 
+	 * - case 1: different lengths => impossible to be permutations of each other 
+	 * - case 2: contains other characters then English alphabet letters
+	 * 
+	 * @param word1
+	 *            first word
+	 * @param word2
+	 *            second word
+	 * @return string describing the error / null
+	 */
+	public static String checkInput(String word1, String word2) {
+		// Check if there is a null string
+		if (word1 == null || word2 == null) {
+			return "Null String";
+		}
+
+		// Check the length
+		if (word1.length() != word2.length()) {
+			return "Different length";
+		}
+
+		// Iterate through each word at the same time
+		for (int i = 0; i < word1.length(); i++) {
+
+			// Check if the current characters are valid
+			if (!isValidCharacter(word1.charAt(i)) || !isValidCharacter(word2.charAt(i))) {
+				return "Invalid input";
+			}
+		}
+
+		// No error
+		return null;

```

Comment:
I didn't think about throwing an Exception, but this is a really good idea! Thank you!